hera_mc/cm_hookup.py

```diff
@@ -79,10 +80,15 @@ def get_hookup(self, hpn_list, rev, port_query='all', at_date='now',
             hookup_dict = self.__get_hookup(hpn_list=self.hookup_list_to_cache, rev='ACTIVE', port_query='all',
                                             at_date=at_date, exact_match=exact_match, show_levels=show_levels)
             self.write_hookup_to_file(hookup_dict)
+            self.hookup_dict_library = copy.copy(hookup_dict)
         else:
-            hookup_dict = self.read_hookup_from_file()
-            if show_levels:
-                hookup_dict = self.__hookup_add_correlator_levels(hookup_dict)
+            if self.hookup_dict_library is None:
+                hookup_dict = self.read_hookup_from_file()
+                if show_levels:
+                    hookup_dict = self.__hookup_add_correlator_levels(hookup_dict)
+                self.hookup_dict_library = copy.copy(hookup_dict)
+            else:
+                hookup_dict = copy.copy(self.hookup_dict_library)
```

Comment:
If the caller of this function is likely to modify the returned `hookup_dict`, then a copy should be used. If it's going to be more or less immutable, then it can be skipped.

Reply:
In fact, the code just below modifies the hookup_dict before it sends it, but we want to keep the full for the next time this gets called.  I would think that all of the copy.copy(ies) would be needed, if any are needed.